todo-list-backend/src/index.ts

```diff
@@ -1,33 +1,208 @@
+    res
+        .status(200)
+        .send(JSON.stringify(todoItems))
+        ;
+    console.log('get todos request returned:: ' + JSON.stringify(todoItems))
+});
+
+app.post("/add-todo", express.json(), (req: Request, res: Response) => {
+    console.log("adding todo")
+
+    if (req.body == undefined) {
+        res
+            .status(400)
+            .send("request body undefined.  sending response code 400")
+            ; console.log("request body undefined.  sending response code 400"); return
+    }
+
+    if (req.body.name == undefined) {
+        res
+            .status(400)
+            .send("task name undefined.  sending response code 400")
+            ; console.log("task name undefined.  sending response code 400"); return
+    }
+
+    let id
+    if (todoItems.length == 0) {
+        id = 0
+    } else {
+        id = todoItems[todoItems.length - 1].id + 1
+    }
+
+    const newTodo = {
+        id: id,
+        name: req.body.name,
+        isDone: false
+    }
+
+    todoItems.push(newTodo)
+
+    saveTodoList(todoItems)
+
     res
-    .setHeader('Access-Control-Allow-Origin', '*')
-    .status(200)
-    .json(responseContent);
+        .status(200)
+        .send(JSON.stringify(newTodo))
+        ; console.log(JSON.stringify(newTodo)); return
 });
 
+app.post("/remove-todo", express.json(), (req: Request, res: Response) => {
+    console.log("removing todo")
+
+    if (req.body == undefined) {
+        res
+            .status(400)
+            .send("request body undefined.  sending response code 400")
+            ; console.log("request body undefined.  sending response code 400"); return
+    }
+
+    if (req.body.id == undefined) {
+        res
+            .status(400)
+            .send("task id undefined.  sending response code 400")
+            ; console.log("task id undefined.  sending response code 400"); return
+    }
+
+    let id = req.body.id
+
+    console.log("removing task id:: " + id)
+
+    let idx = todoItems.findIndex(function (item) { return item.id == id })
+
+    if (idx == -1) {
+        res
+            .status(400)
+            .send("no element in list of requested index")
+            ; console.log("no element in list of requested index"); return
+    }
+
+    let removedElement = todoItems.splice(idx, 1)[0]
+
+    res
+        .status(200)
+        .send(JSON.stringify(removedElement))
+});
+
+app.post("/tag-task-as-complete", express.json(), (req: Request, res: Response) => {
+    console.log("tagging task as complete")
+    const request = req.body;
+    const id = request.id
+    const strict = request.strict;
+    let result
+    try {
+        result = changeCompletionStatusHandler(id, strict, true)
+    } catch (e) {
+        if (e instanceof Error) {
+            res
+                .status(400)
+                .send(e.message)
+                ;
+        } else {
+            res
+                .status(500)
+                .send("unknown backend error")
+        }
```

Comment:
This check is likely unnecessary, for two reasons:
- Even internal errors like program crashes will still be of the `Error` type - not only the ones you explicitly throw, so 500-class errors wouldn't actually end up here
- If we're throwing something other than an `Error`, we've already strayed too far from God's light - it's probably not something you should have to account for.